packages/medusa/src/services/cart.ts

```diff
@@ -313,6 +313,7 @@ class CartService extends TransactionBaseService {
 
         const rawCart: DeepPartial<Cart> = {
           context: data.context ?? {},
+          metadata: data.metadata ?? {}
```

Comment:
todo: this can be removed, it is manage l418-431

Reply:
suggestion: rm line 421 and after before the remaining fields do a check if data.metadata is present and use the setMetadata utils like in the other places

Reply:
Done ✅
Thanks for the suggestion.